DSA Session/Assignment6/Graph Operations/src/graph/GraphImpl.java

```diff
@@ -0,0 +1,297 @@
+package graph;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import pojo.Edge;
+
+public class GraphImpl implements Graph {
+    List<List<Edge>> edgeList;
+    int vertices;
+    List<Edge> listOfEdgesInMST;
+
+    /**
+     * Constructs the graph with the given number of vertices.
+     * @param vertices : number of vertices.
+     */
+    public GraphImpl(int vertices) {
+        try {
+            if (vertices <= 0) {
+                throw new Exception("Invalid value of vertices");
+            }
+            this.vertices = vertices;
+            edgeList = new ArrayList<>(vertices);
+            for (int i = 0; i < vertices; i++) {
+                edgeList.add(new LinkedList<Edge>());
+            }
+            listOfEdgesInMST = new ArrayList<Edge>();
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+    public boolean addEdge(int source, int destination, int weight) {
```

Comment:
Please do documentation for methods.

Reply:
Sir, I have done it in the interface, that's why I did not do it here.